src/Raven.Pal/src/posix/spawn.c

```diff
@@ -0,0 +1,291 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <sys/wait.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <assert.h>
+#include <time.h>
+#include <string.h>
+#include <signal.h>
+
+#include "rvn.h"
+#include "status_codes.h"
+#include "internal_posix.h"
+
+
+#define MSEC_TO_SEC(x) (x/1000)
+#define MSEC_TO_NANOSEC(x) ((x % 1000) * 1000000)
+
+PRIVATE char** 
+parse_cmd_line(char* line)
+{
+    if (line == NULL || *line == '\0')
+        return NULL;
+
+    char** argv = malloc(strlen(line)*2);
```

Comment:
We don't actually need this size, do we? Let's default to `malloc(sizeof(char**) * 8)` and `realloc` as needed?